### PR TITLE
Add support for styled-jsx external styles

### PR DESCRIPTION
**Pull request by hongrich**
Supports all 3 tags in `styled-jsx/css`: `css`, `css.global`, and `css.resolve`.
https://github.com/zeit/styled-jsx/#external-css-and-styles-outside-of-the-component

Fixes https://github.com/zeit/styled-jsx/issues/548

Formatting template literal with `css` was already working because it
happens to be the same as what styled-components was using.
https://www.styled-components.com/docs/api#css

So this change just adds support for `css.global` and `css.resolve`.

- [x] I’ve added tests to confirm my change works.
- [x] (If the change is user-facing) I’ve added my changes to the `CHANGELOG.unreleased.md` file following the template.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by hongrich** [comment]
I rebased this change on top the latest and added a change log entry. Let me know if there's anything else that should be done before this can be merged.

Thank you!

**Comment by duailibe** [comment]
This *seems* safe, but I'm more and more afraid of formatting template literals

**Comment by evilebottnawi** [comment]
@duailibe i think we should disable this by default in 2 version

**Comment by hongrich** [comment]
Just rebased on top the latest, so should be mergeable again. This seem like a small enough change that should hopefully be pretty safe. It'd certainly help the developer experience in working with [`styled-jsx`](https://github.com/zeit/styled-jsx/issues/548).

I agree that it'd be good to separately take a closer look at formatting template literals as a whole for Prettier 2.0

**Comment by duailibe** [comment]
@lydell @evilebottnawi Since this only applies to `css.global` and `css.resolve` it seems specific enough to be "safe". I'm willing to merge this and will start to think a solution for this problem for future versions.

WDYT?

**Comment by lydell** [comment]
Let's do it!

File: src/language-js/embed.js [hunk 1]
```diff
@@ -335,22 +335,35 @@ function printGraphqlComments(lines) {
...
+      parent.type === "JSXExpressionContainer" &&
+      parentParent.type === "JSXElement" &&
+      parentParent.openingElement.name.name === "style" &&
+      parentParent.openingElement.attributes.some(
+        attribute => attribute.name.name === "jsx"
+      )) ||
+    (parent &&
+      parent.type === "TaggedTemplateExpression" &&
+      parent.tag.type === "Identifier" &&
+      parent.tag.name === "css") ||
```
Comment (duailibe) [inline src/language-js/embed.js]: FTR ``` css`` ``` was already supported because of styled components

**Comment by duailibe** [comment]
The ``` css`` ``` check is now duplicated... I'll merge this as-is and refactor

**Pull request merged**